Examples/CMakeLists.txt

```diff
@@ -72,14 +72,25 @@ function(add_warpx_test
     separate_arguments(ANALYSIS_LIST UNIX_COMMAND "${analysis}")
     list(GET ANALYSIS_LIST 0 ANALYSIS_FILE)
     cmake_path(SET ANALYSIS_FILE "${CMAKE_CURRENT_SOURCE_DIR}/${ANALYSIS_FILE}")
-    # TODO Enable lines below to handle command-line arguments
-    #list(LENGTH ANALYSIS_LIST ANALYSIS_LIST_LENGTH)
-    #if(ANALYSIS_LIST_LENGTH GREATER 1)
-    #    list(SUBLIST ANALYSIS_LIST 1 -1 ANALYSIS_ARGS)
-    #    list(JOIN ANALYSIS_ARGS " " ANALYSIS_ARGS)
-    #else()
-    #    set(ANALYSIS_ARGS "")
-    #endif()
+    list(LENGTH ANALYSIS_LIST ANALYSIS_LIST_LENGTH)
+    if(ANALYSIS_LIST_LENGTH GREATER 1)
+        list(SUBLIST ANALYSIS_LIST 1 -1 ANALYSIS_ARGS)
+        list(JOIN ANALYSIS_ARGS " " ANALYSIS_ARGS)
+    else()
+        set(ANALYSIS_ARGS "")
+    endif()
+
+    # get checksum script and optional command-line arguments
+    separate_arguments(CHECKSUM_LIST UNIX_COMMAND "${checksum}")
+    list(GET CHECKSUM_LIST 0 CHECKSUM_FILE)
```

Comment:
I'm thinking about ways to make this simpler. Instead of having a soft link each place where the default analysis script is used, can this check if `CHECKSUM_FILE` is equal to `analysis_default_regression.py` and put in the appropriate direct path? Perhaps use some keyword like "DEFAULT" instead of the script name?

Reply:
I guess that would be nice, but when Axel and I set up ctest in the first place we did not manage to make it work with the direct paths (there were several attempts and then we turned to soft links). Especially, I think Axel found issues with running ctest from within an IDE. We could rivisit once he is back, but I would not try this here.

Reply:
Thinking further, instead of writing out the full command to do the benchmark check, could the arguments to the analysis script be added as arguments to this function? In most cases, default values could be used so most `CMakeLists.txt` files wouldn't need any extra arguments (especially with the automatic discovery I suggest above). Then the `rtol` and `skip-fields` and `skip-particles` could be added via variable arguments pairs, like this
```
rtol 1.e-5
```

Reply:
I think this was another design choice that we made with Axel initially. 

We wanted to keep the number of arguments of `add_warpx_test` as little as possible, especially given that we cannot use keyword arguments like in Python. 

Another issue I see is that we would be treating the arguments to the "custom" analysis script (which cannot be predicted in advance) differently than the arguments to the default analysis script.

If you feel strongly about this, I would recommend that we wait until Axel is back and discuss with him further.

Reply:
Another comment is that we wanted to display commands that developers could copy/paste to try things locally, whenever that is needed (e.g., for debugging purposes). This came out of discussions with @aeriforme, and I think it would be valuable in general, as opposed to passing arguments to `add_warpx_test` even if they are actually arguments read/used by the analysis scripts.

Reply:
I don't feel strongly about this, just brain storming ideas. I see that having the command to do the benchmark check written out is convenient for debugging.

BTW, you can imitate keyword arguments though it's a bit kludgy. You can use variable arguments and scan through them. If one has a value of the argument name, then take the next argument as the value.